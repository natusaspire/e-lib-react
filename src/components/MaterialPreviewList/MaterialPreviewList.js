import React, { PureComponent } from 'react';

import {
  MATERIALS_TEXT_ENDPOINT,
  MATERIALS_AUDIO_ENDPOINT,
  MATERIALS_VIDEO_ENDPOINT
} from '@/constants/endpoints';
import MaterialPreview from '@/components/MaterialPreview';
import textImage from '@/assets/categories/text.png';
import audioImage from '@/assets/categories/audio.png';
import videoImage from '@/assets/categories/video.png';

const IMAGES = new Map()
  .set(MATERIALS_TEXT_ENDPOINT, textImage)
  .set(MATERIALS_AUDIO_ENDPOINT, audioImage)
  .set(MATERIALS_VIDEO_ENDPOINT, videoImage);

class MaterialPreviewList extends PureComponent {
  renderMaterial = material => {
    const { endpoint } = this.props;

    return (
      <div key={material.id} className="grid__item">
        <MaterialPreview
          endpoint={endpoint}
          material={material}
          image={IMAGES.get(endpoint)}
        />
      </div>
    );
  };

  render() {
    const { materials } = this.props;

    return (
      <div className="grid content">
        {materials.map(this.renderMaterial)}
      </div>
    );
  }
}

export default MaterialPreviewList;
