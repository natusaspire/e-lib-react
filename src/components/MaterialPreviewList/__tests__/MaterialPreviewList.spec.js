import React from 'react';
import { shallow } from 'enzyme';

import MaterialPreviewList from '@/components/MaterialPreviewList';

jest.mock('@/constants/endpoints', () => ({
  MATERIALS_TEXT_ENDPOINT: 'text',
  MATERIALS_AUDIO_ENDPOINT: 'audio',
  MATERIALS_VIDEO_ENDPOINT: 'video'
}));

jest.mock('@/components/MaterialPreview', () => 'MaterialPreview');
jest.mock('@/assets/categories/text.png', () => 'text.png');
jest.mock('@/assets/categories/audio.png', () => 'audio.png');
jest.mock('@/assets/categories/video.png', () => 'video.png');

describe('<MaterialPreviewList />', () => {
  it('should render correctly when endpoint is "text"', () => {
    const props = {
      endpoint: 'text',
      materials: [
        {
          id: 1,
          type: 'text',
          title: 'Material title 1',
          textContent: 'Material text content 1.'
        },
        {
          id: 2,
          type: 'text',
          title: 'Material title 2',
          textContent: 'Material text content 2.'
        },
        {
          id: 3,
          type: 'text',
          title: 'Material title 3',
          textContent: 'Material text content 3.'
        }
      ]
    };

    const wrapper = shallow(<MaterialPreviewList {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when endpoint is "audio"', () => {
    const props = {
      endpoint: 'audio',
      materials: [
        {
          id: 1,
          type: 'audio',
          title: 'Material title 1',
          url: 'audio1.mp3'
        },
        {
          id: 2,
          type: 'audio',
          title: 'Material title 2',
          url: 'audio2.mp3'
        },
        {
          id: 3,
          type: 'audio',
          title: 'Material title 3',
          url: 'audio3.mp3'
        }
      ]
    };

    const wrapper = shallow(<MaterialPreviewList {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when endpoint is "video"', () => {
    const props = {
      endpoint: 'video',
      materials: [
        {
          id: 1,
          type: 'video',
          title: 'Material title 1',
          url: 'video1.mp3'
        },
        {
          id: 2,
          type: 'video',
          title: 'Material title 2',
          url: 'video2.mp3'
        },
        {
          id: 3,
          type: 'video',
          title: 'Material title 3',
          url: 'video3.mp3'
        }
      ]
    };

    const wrapper = shallow(<MaterialPreviewList {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
