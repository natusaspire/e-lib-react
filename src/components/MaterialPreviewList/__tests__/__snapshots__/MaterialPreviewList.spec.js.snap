// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MaterialPreviewList /> renderMaterial should render material 1`] = `
<div
  className="grid__item"
>
  <MaterialPreview
    endpoint="text"
    image="text.png"
    material={
      Object {
        "id": 1,
        "textContent": "Material text content 1.",
        "title": "Material title 1",
        "type": "text",
      }
    }
  />
</div>
`;

exports[`<MaterialPreviewList /> should render audio materials when endpoint is "audio" 1`] = `
<div
  className="grid content"
>
  <div
    className="grid__item"
    key="1"
  >
    <MaterialPreview
      endpoint="audio"
      image="audio.png"
      material={
        Object {
          "id": 1,
          "title": "Material title 1",
          "type": "audio",
          "url": "audio1.mp3",
        }
      }
    />
  </div>
  <div
    className="grid__item"
    key="2"
  >
    <MaterialPreview
      endpoint="audio"
      image="audio.png"
      material={
        Object {
          "id": 2,
          "title": "Material title 2",
          "type": "audio",
          "url": "audio2.mp3",
        }
      }
    />
  </div>
  <div
    className="grid__item"
    key="3"
  >
    <MaterialPreview
      endpoint="audio"
      image="audio.png"
      material={
        Object {
          "id": 3,
          "title": "Material title 3",
          "type": "audio",
          "url": "audio3.mp3",
        }
      }
    />
  </div>
</div>
`;

exports[`<MaterialPreviewList /> should render text materials when endpoint is "text" 1`] = `
<div
  className="grid content"
>
  <div
    className="grid__item"
    key="1"
  >
    <MaterialPreview
      endpoint="text"
      image="text.png"
      material={
        Object {
          "id": 1,
          "textContent": "Material text content 1.",
          "title": "Material title 1",
          "type": "text",
        }
      }
    />
  </div>
  <div
    className="grid__item"
    key="2"
  >
    <MaterialPreview
      endpoint="text"
      image="text.png"
      material={
        Object {
          "id": 2,
          "textContent": "Material text content 2.",
          "title": "Material title 2",
          "type": "text",
        }
      }
    />
  </div>
  <div
    className="grid__item"
    key="3"
  >
    <MaterialPreview
      endpoint="text"
      image="text.png"
      material={
        Object {
          "id": 3,
          "textContent": "Material text content 3.",
          "title": "Material title 3",
          "type": "text",
        }
      }
    />
  </div>
</div>
`;

exports[`<MaterialPreviewList /> should render video materials when endpoint is "video" 1`] = `
<div
  className="grid content"
>
  <div
    className="grid__item"
    key="1"
  >
    <MaterialPreview
      endpoint="video"
      image="video.png"
      material={
        Object {
          "id": 1,
          "title": "Material title 1",
          "type": "video",
          "url": "video1.mp3",
        }
      }
    />
  </div>
  <div
    className="grid__item"
    key="2"
  >
    <MaterialPreview
      endpoint="video"
      image="video.png"
      material={
        Object {
          "id": 2,
          "title": "Material title 2",
          "type": "video",
          "url": "video2.mp3",
        }
      }
    />
  </div>
  <div
    className="grid__item"
    key="3"
  >
    <MaterialPreview
      endpoint="video"
      image="video.png"
      material={
        Object {
          "id": 3,
          "title": "Material title 3",
          "type": "video",
          "url": "video3.mp3",
        }
      }
    />
  </div>
</div>
`;
